Findcppnetlib.cmake

```diff
@@ -0,0 +1,93 @@
+# contributors may be used to endorse or promote products derived from this
+# software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+# =============================================================================
+#
+# CPPNETLIB 
+# 
+#==============================================================================
+#
+# This module defines the following output variables:
+#
+# CPPNETLIB_FOUND - Was cppnetlib and all of the specified components found?
+#
+# CPPNETLIB_INCLUDE_DIRS - Where to find the headers
+#
+# CPPNETLIB_LIBRARIES - The cppnetlib libraries
+#==============================================================================
+#
+
+# Assume not found.
+SET(CPPNETLIB_FOUND FALSE)
+
+# PATH ________________________________________________________________________
+
+if (NOT CPPNETLIB_PATH)
+    find_path(CPPNETLIB_PATH include/boost/network.hpp
+	HINTS ${CPP_NETLIB_ROOT} $ENV{CPP_NETLIB_ROOT}
+        /usr/local/
+        /usr/
+    )
+endif ()
+
+# HEADERS AND DYNAMIC LIBRARIES_________________________________________________
+
+if (CPPNETLIB_PATH)
+    set (CPPNETLIB_INCLUDE_DIR ${CPPNETLIB_PATH}/include)
+    mark_as_advanced (CPPNETLIB_INCLUDE_DIR)
+
+    find_library(CPPNETLIB_LIBRARY1 NAMES cppnetlib-client-connections
+	HINTS ${CPPNETLIB_ROOT} $ENV{CPPNETLIB_ROOT}
+        PATHS ${CPPNETLIB_PATH}/lib64
+    )
+    find_library(CPPNETLIB_LIBRARY2 NAMES cppnetlib-server-parsers
+        HINTS ${CPPNETLIB_ROOT} $ENV{CPPNETLIB_ROOT}
+        PATHS ${CPPNETLIB_PATH}/lib64
+    )
+    find_library(CPPNETLIB_LIBRARY3 NAMES cppnetlib-uri
+        HINTS ${CPPNETLIB_ROOT} $ENV{CPPNETLIB_ROOT}
+        PATHS ${CPPNETLIB_PATH}/lib64
+    )
+    SET(CPPNETLIB_LIBRARIES ${CPPNETLIB_LIBRARY1} ${CPPNETLIB_LIBRARY2} ${CPPNETLIB_LIBRARY3})
+    mark_as_advanced(CPPNETLIB_LIBRARIES)
+endif ()
+
+# FOUND _______________________________________________________________________
+if (CPPNETLIB_FIND_REQUIRED)
+    set(_cppnetlib_output_type FATAL_ERROR)
+else()
+    set(_cppnetlib_output_type STATUS)
+endif ()
+
+include(FindPackageHandleStandardArgs)
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(CPPNETLIB DEFAULT_MSG CPPNETLIB_LIBRARIES CPPNETLIB_INCLUDE_DIR)
```

Comment:
no need for UPPER_CASE function name. Lower-case also works and looks less aggressive :)